demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js

```diff
@@ -1,16 +1,86 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {Carousel, Image} from 'react-bootstrap';
 import {Link} from 'react-router-dom';
 import slideOne from '../../../images/slide_one.png';
 import slideTwo from '../../../images/slide_two.png';
 import slideThree from '../../../images/slide_three.png';
+import {event} from '../../../lib/gtag.js';
+import {getItemParameters} from '../../../utils.js';
 import './HomeCarousel.css';
 
 /**
  * Creates slideshow component for showcasing our latest deals and offerings.
- * @return {!JSX} The component.
+ * @return {!JSX}
  */
 export function HomeCarousel() {
+  // Send view_promotion event when the carousel loads for the first time.
+  useEffect(() => {
+    event('view_promotion', {
+      items: [
+        {
+          ...getItemParameters('7ba94'),
+          promotion_name: 'Home Carousel',
+        },
+        {
+          ...getItemParameters('hjdf7'),
+          promotion_name: 'Home Carousel',
+          discount: 25,
+        },
+        {
+          ...getItemParameters('3h488'),
+          promotion_name: 'Home Carousel',
+        },
+      ],
+      promotion_name: 'Home Carousel',
+    });
+  }, []);
+
+  /**
+   * Send select_promotion event to Google Analytics when the first
+   * carousel slide item is selected.
+   */
+  function onSlideOneClick() {
```

Comment:
There's a lot of redundancy between the different onSlideXClick methods here.

WDYT about refactoring to a single `onSlideClicked(itemId, discount = undefined)` and passing arguments  in the onClick handlers?
e.g. https://reactjs.org/docs/handling-events.html#passing-arguments-to-event-handlers

Reply:
This is a good idea, I definitely felt bad about all the copy + pasting! I'll update the PR asap